util/devel/grepstdchdrs

```diff
@@ -16,4 +16,4 @@ fi
 
 filter=${1:+cat}
 cd $project_dir && \
-find compiler \( \( -name '*.cpp' -o -name '*.h' \) -a ! -name 'chapel.tab.cpp' -a ! -name 'lex.yy.cpp' \) -exec egrep ${1:--s} '^[       ]*#[    ]*include[      ][      ]*<[    ]*(assert|ctype|errno|float|iso646|limits|locale|math|setjmp|signal|stdarg|stddef|stdio|stdlib|string|time|wchar|wctype)[.]h[   ]*>' {} \; -print | `echo "${filter:-grep -v include[^/]}"`a
+find compiler \( \( -name '*.cpp' -o -name '*.h' \) -a ! -name 'bison-chapel.cpp' -a ! -name 'flex-chapel.cpp' -a ! -name 'flex-chapel.h' \) -exec egrep ${1:--s} '^[       ]*#[    ]*include[      ][      ]*<[    ]*(assert|ctype|errno|float|iso646|limits|locale|math|setjmp|signal|stdarg|stddef|stdio|stdlib|string|time|wchar|wctype)[.]h[   ]*>' {} \; -print | `echo "${filter:-grep -v include[^/]}"`a
```

Comment:
I considered that.  Historically this script didn’t check either header; there wasn’t a flex header and,
apparently, the bison header doesn’t cause trouble.  The new flex header is causing trouble and
so I added it.  Perhaps the bison header will cause trouble one day but I liked the idea of leaving
things as close to current expectations as I reasonably could.

Mike


On Feb 17, 2015, at 2:49 PM, Lydia Duncan <notifications@github.com<mailto:notifications@github.com>> wrote:


In util/devel/grepstdchdrs<https://github.com/chapel-lang/chapel/pull/1331#discussion_r24862760>:

> @@ -16,4 +16,4 @@ fi
>
>  filter=${1:+cat}
>  cd $project_dir && \
> -find compiler \( \( -name '*.cpp' -o -name '*.h' \) -a ! -name 'chapel.tab.cpp' -a ! -name 'lex.yy.cpp' \) -exec egrep ${1:--s} '^[       ]*#[    ]*include[      ][      ]*<[    ]*(assert|ctype|errno|float|iso646|limits|locale|math|setjmp|signal|stdarg|stddef|stdio|stdlib|string|time|wchar|wctype)[.]h[   ]*>' {} \; -print | `echo "${filter:-grep -v include[^/]}"`a
> +find compiler \( \( -name '*.cpp' -o -name '*.h' \) -a ! -name 'bison-chapel.cpp' -a ! -name 'flex-chapel.cpp' -a ! -name 'flex-chapel.h' \) -exec egrep ${1:--s} '^[       ]*#[    ]*include[      ][      ]*<[    ]*(assert|ctype|errno|float|iso646|limits|locale|math|setjmp|signal|stdarg|stddef|stdio|stdlib|string|time|wchar|wctype)[.]h[   ]*>' {} \; -print | `echo "${filter:-grep -v include[^/]}"`a


I notice you check against the flex-chapel.h file but not the bison-chapel.h file. Why one but not the other?

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/1331/files#r24862760>.